src/app/app.component.ts

```diff
@@ -17,10 +17,18 @@ export class AppComponent implements OnInit {
   loading!: Observable<boolean>;
 
   constructor(
-    private loadingService: LoadingService
+    private loadingService: LoadingService,
+    private authService: AuthenticationService
   ) {}
+
   
   ngOnInit(): void {
     this.loading = this.loadingService.loadingSubject.asObservable();
```

Comment:
Please move this logic to the service
Make the subject private
And create the getter of it - and return subject as Observable from it

Reply:
done. Besides the getter I needed to create also the setter when I will change the loading status of loadingSubject in components.